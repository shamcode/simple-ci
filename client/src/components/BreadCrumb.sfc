<template>
    {% import path from 'sham-ui-router/lib/params' %}

    <ul class="breadcrumb">
        <li>
            <a :hrefto={{path("project-list")}}>
                Projects
            </a>
        </li>
        {% if project && project.id %}
            <li>
                <a :hrefto={{path("project-detail").param("id", project.id)}}>
                    {{project.name}}
                </a>
            </li>
            {% if chain && chain.id %}
                <li>
                    <a :hrefto={{path("project-chain-detail").param("id", project.id).param("chainId", chain.id)}}>
                        {{chain.name}}
                    </a>
                </li>
            {% endif %}
        {% endif %}
        <li>
            {% if this[ ref( 'dataReady' ) ]( project, chain ) %}
                {{currentPage}}
            {% endif %}
        </li>
    </ul>
</template>

<script>
    function BreadCrumb( options ) {
        const project = ref();
        const chain = ref();
        const currentPage = ref();

        options( {
            [ project ]: null,
            [ chain ]: null,
            [ currentPage ]: ''
        } );

        this[ ref( 'dataReady' ) ] = ( project, chain ) => (
            ( project === null || project.id ) &&
            ( chain === null || chain.id )
        );
    }

    export default Component( Template, BreadCrumb );
</script>
