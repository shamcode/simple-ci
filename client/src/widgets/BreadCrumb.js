import { options } from 'sham-ui';
import BreadCrumbTemplate from './BreadCrumb.sht';

export default class BreadCrumb extends BreadCrumbTemplate {
    @options get project() { return null }
    @options get chain() { return null }
    @options get currentPage() { return ''; }

    routerParamsForProject( project ) {
        return {
            id: project === undefined ? null : project.id
        }
    }

    routerParamsForChain( project, chain ) {
        return {
            id: project === undefined ? null : project.id,
            chainId: chain === undefined ? null : chain.id
        }
    }

    dataReady( project, chain ) {
        return ( project === null || project.id ) && ( chain === null || chain.id )
    }
}