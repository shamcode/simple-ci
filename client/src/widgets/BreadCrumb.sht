{% import LinkTo from 'sham-ui-router/link-to' %}

<ul class="breadcrumb">
    <li>
        <LinkTo
                text="Projects"
                path="project-list"
        />
    </li>
    {% if project && project.id %}
    <li>
        <LinkTo
                text={{project.name}}
                path="project-detail"
                params={{this.routerParamsForProject(project)}}
        />
    </li>
    {% endif %}
    {% if chain && chain.id %}
    <li>
        <LinkTo
                text={{chain.name}}
                path="project-chain-detail"
                params={{this.routerParamsForChain(project, chain)}}
        />
    </li>
    {% endif %}
    <li>
        {% if this.dataReady( project, chain ) %}
            {{currentPage}}
        {% endif %}
    </li>
</ul>