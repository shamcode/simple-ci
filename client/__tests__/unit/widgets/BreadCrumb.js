import { DI } from 'sham-ui';
import BreadCrumb from '../../../src/widgets/BreadCrumb';
import renderer from 'sham-ui-test-helpers';

afterEach( () => {
    DI.bind( 'router', null );
} );

it( 'renders correctly', () => {
    DI.bind( 'router', {
        generate: jest.fn().mockReturnValueOnce( '/' )
    } );

    const meta = renderer( BreadCrumb, {} );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
