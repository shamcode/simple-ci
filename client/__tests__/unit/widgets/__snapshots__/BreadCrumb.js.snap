// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Constructor": "BreadCrumb",
  "Options": Object {
    "ID": "widget",
    "chain": null,
    "containerSelector": "body",
    "currentPage": "",
    "project": null,
    "types": Array [],
  },
  "html": "
<ul class=\\"breadcrumb\\">
  <li>
    <a href=\\"/\\" class=\\"\\">Projects
      <!--default-->
    </a>
  </li>
  <!--if-->
  <!--if-->
  <li></li>
</ul>
",
}
`;
